pkg/operator/podidentity/podidentitywebhook_controller_test.go

```diff
@@ -0,0 +1,317 @@
+				clientset:      fakeClientset,
+				logger:         logger,
+				eventRecorder:  events.NewInMemoryRecorder(""),
+				cache:          resourceapply.NewResourceCache(),
+				conditions:     []configv1.ClusterOperatorStatusCondition{},
+				imagePullSpec:  podIdentityObj.GetImagePullSpec(),
+				podIdentityObj: podIdentityObj,
+			}
+
+			_, err := r.Reconcile(context.TODO(), reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "testName",
+					Namespace: "testNamespace",
+				},
+			})
+			switch {
+
+			}
+
+			if err != nil && !test.expectErr {
+				require.NoError(t, err, "Unexpected error: %v", err)
+			}
+			if err == nil && test.expectErr {
+				t.Errorf("Expected error but got none")
+			}
+
+			if !test.expectErr {
+				expectedImage := fmt.Sprintf("%s_identity_image", strings.ToLower(string(test.platformType)))
+				assert.Equal(t, r.imagePullSpec, expectedImage)
+
+				podIdentityWebhookDeployment := getDeployment(fakeClientset, "pod-identity-webhook", "openshift-cloud-credential-operator")
+				assert.NotNil(t, podIdentityWebhookDeployment, "did not find expected pod-identity-webhook Deployment")
+
+				if test.expectedReplicas != 0 {
+					assert.Equal(t, *podIdentityWebhookDeployment.Spec.Replicas, test.expectedReplicas, "found unexpected pod-identity-webhook deployment replicas")
+				}
+
+				assert.Equal(t, podIdentityWebhookDeployment.Spec.Template.Spec.Containers[0].Image, expectedImage, "container image matches expected one")
+
+				podDisruptionBudget := getPDB(fakeClientset, "pod-identity-webhook", "openshift-cloud-credential-operator")
+				if test.expectPDB {
+					assert.NotNil(t, podDisruptionBudget, "did not find expected pod-identity-webhook PodDisruptionBudget")
+				} else {
+					assert.Nil(t, podDisruptionBudget, "found unexpected pod-identity-webhook PodDisruptionBudget")
+				}
+
+				webhook := getWebhook(fakeClientset, "pod-identity-webhook")
+				assert.NotNil(t, webhook, "did not find expected pod-identity-webhook webhook config")
+				assert.Contains(t, webhook.Webhooks[0].Name, strings.ToLower(string(test.platformType)))
+			}
+		})
+	}
+}
+
+func TestPodIdentityShouldDeploy(t *testing.T) {
+	schemeutils.SetupScheme(scheme.Scheme)
+
+	azureCredsSecretName := "azure-credentials"
+	tests := []struct {
+		name           string
+		existing       []runtime.Object
+		expectedResult bool
+		expectErr      bool
+		podIdentityObj PodIdentityInterface
+	}{{
+		name: "Azure secret with credentials exists",
+		existing: []runtime.Object{
+			&corev1.Secret{
+				ObjectMeta: v1.ObjectMeta{
+					Name: azureCredsSecretName,
+				},
+				Data: map[string][]byte{
+					azureTenantIdKey: {},
+				},
+			}},
+		podIdentityObj: AzurePodIdentity{},
+		expectedResult: true,
+		expectErr:      false,
+	},
+		{
+			name: "Azure secret with credentials without tenant id",
+			existing: []runtime.Object{
+				&corev1.Secret{
+					ObjectMeta: v1.ObjectMeta{
+						Name: azureCredsSecretName,
```

Comment:
These fixtures don't have a `namespace` - why does this work?

Reply:
As i don't provide it to shouldBeDeployed, i will add namespace in case it helps